docs/design/proposals/rgd-ownership-and-deletion.md

```diff
@@ -0,0 +1,203 @@
+
+Together, these changes align KRO with Kubernetes-native resource management and improve data safety.
+
+---
+
+### Design details
+
+#### 1. Ownership introduction
+
+Each CRD created by an RGD would carry an **OwnerReference** set by the controller:
+
+```go
+if err := ctrl.SetControllerReference(rgd, crd, r.Scheme()); err != nil {
+    mark.KindUnready(err.Error())
+    return nil, nil, fmt.Errorf("failed to set controller reference of CRD: %w", err)
+}
+```
+
+This produces CRDs with metadata similar to:
+
+```yaml
+metadata:
+  name: sample.kro.run
+  ownerReferences:
+    - apiVersion: kro.run/v1alpha1
+      kind: ResourceGraphDefinition
+      name: sample
+      uid: 12345-abcde
+      controller: true
+      blockOwnerDeletion: true
+```
+
+**Resulting behavior:**
+
+* When an RGD is deleted, its owned CRD is automatically garbage-collected. This is proper behavior
+  as we can still delete RGDs with policy "orphan" to preserve CRDs.
+* Manual CRD deletion (while RGD still exists) will trigger re-reconciliation, as ownership implies controller responsibility.
+  (we can and should think about warning or failing user deletion requests on CRDs managed by KRO)
+* The original RGD label remains for traceability, but lifecycle is now managed by native Kubernetes GC.
+
+Integration tests confirm the presence and correctness of `ownerReferences` (including `controller` and `blockOwnerDeletion` flags).
+
+#### 2. Deletion protection for RGDs
+
+Because RGD deletion now cascades to delete its CRD, an **admission policy** prevents accidental data loss.
+
+When `config.allowCRDDeletion` is `false` (default), Helm installs a **ValidatingAdmissionPolicy** and **Binding**:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: {{ include "kro.fullname" . }}-crd-protection-policy
+spec:
+  matchConstraints:
+    resourceRules:
+      - apiGroups: ["kro.run"]
+        apiVersions: ["v1alpha1"]
+        operations: ["DELETE"]
+        resources: ["resourcegraphdefinitions"]
+  validations:
+    - expression: |
+        has(oldObject.metadata.annotations) &&
+        oldObject.metadata.annotations['{{ .Values.validation.admission.policy.rgd.annotation.key }}'] == 'true'
+      reason: Invalid
+      message: |
+        Deletion denied. To proceed, set annotation '{{ .Values.validation.admission.policy.rgd.annotation.key }}: "true"'.
+        Removing an RGD also deletes its CustomResourceDefinition and may cause data loss.
+```
+
+Users must explicitly annotate an RGD to confirm deletion:
+
+```yaml
+metadata:
+  annotations:
+    kro.run/allow-delete: "true"
+```
+
+This ensures intentional deletions only.
+
+#### 3. Controller-side deletion control (`allowCRDDeletion`)
+
+Previously, the flag `allowCRDDeletion` determined whether KRO would delete CRDs it had created if the RGD was removed or changed.
```

Comment:
I'm curious to explore the use cases for setting this flag one way or another. If we're certain that all of the instances managed by the RGD are gone, is it really dangerous to remove the Kind? I'm not seeing the upsides of leaking it if we're confident we're not losing data. I can see downsides of not removing the kind, such as attempting to do a fresh installation of a new version where the Kind has changed substantially.